group-coordinator/src/main/java/org/apache/kafka/coordinator/group/GroupMetadataManager.java

```diff
@@ -6280,6 +6286,48 @@ public void createGroupTombstoneRecords(
         group.createGroupTombstoneRecords(records);
     }
 
+    /**
+     * Returns an optional of delete share group request object to be used with the persister.
```

Comment:
This doesn't really do the method justice. It seems to me that the salient points are:
* It checks whether the share group is empty. If it is not empty, the return value is empty.
* If there are no subscribed topics, again the return value is empty.
* Otherwise, it returns the parameters required by the persister to delete the share-partition state.

I would rename the method to something like `shareGroupCheckEmptyAndBuildPartitionDeleteRequest`.

Reply:
It could even throw `GroupNotEmptyException` if the group is not empty. That seems like the cleanest way out of this deletion process for a non-empty share group.

Reply:
`shareGroupCheckEmptyAndBuildPartitionDeleteRequest` - thats a mouthful

Reply:
Ok - the empty check actually does not belong here. I will move the check to the `GroupCoordinatorShard.sharePartitionDeleteRequests`